### ISSUE TITLE
Problem connecting to UDP server

### ISSUE THREAD

**Issue by pml-jpe** [M0r13n/pyais#33]
Hi all, I am trying to use pyais to read a UDP output from an AIS receiver on the network. I can read it fine using `netcat`:

```
jpe@pmpc1770:~>nc -vu 192.171.162.68 12345
Connection to 192.171.162.68 12345 port [udp/*] succeeded!
\s:PML,c:1622632119,t:LIVE*10\!AIVDM,1,1,,A,13P:DSP000wdk1vLm2mRa@C:08GI,0*74
\s:PML,c:1622632119,t:LIVE*10\!AIVDM,1,1,,B,13M@LCP000wdm98LmIidL7u<0L0M,0*6C
\s:PML,c:1622632119,t:LIVE*10\!AIVDM,1,1,,A,13:dR<5Ohkwe;BDLlES`=Fo>0<0Q,0*53
\s:PML,c:1622632119,t:LIVE*10\!AIVDM,1,1,,B,139K?fP00ewe2jLLe7C01P1>00T1,0*59
\s:PML,c:1622632119,t:LIVE*10\!AIVDM,1,1,,B,339K?i001;wdoV:LjCuW9Ec@2000,0*01
\s:PML,c:1622632120,t:LIVE*1A\!AIVDM,1,1,,B,13MAPkPP00OdoipLl@E>4?wB2D08,0*16
\s:PML,c:1622632121,t:LIVE*1B\!AIVDM,1,1,,A,13M@ENh000wdn9RLm76<BUG>08Hb,0*3C
\s:PML,c:1622632121,t:LIVE*1B\!AIVDM,1,1,,B,13M@E50000OdfwlLmKBSC@sD08Hl,0*6F
\s:PML,c:1622632122,t:LIVE*18\$AIHBT,5.0,A,8*28
\s:PML,c:1622632122,t:LIVE*18\!AIVDM,1,1,,B,15QJmN002;OehoJLU5rprW;H0<0t,0*47
\s:PML,c:1622632122,t:LIVE*18\!AIVDM,1,1,,A,13OhOf0P?w<tSF0l4Q@>4?wv0PS7,0*16
\s:PML,c:1622632122,t:LIVE*18\!AIVDM,1,1,,B,13P<SaS000wdiKhLmEFPhlIF0@I`,0*63
\s:PML,c:1622632123,t:LIVE*19\!AIVDM,1,1,,B,13P;r00000wdmFlLmLCf42CF2@Ij,0*40
\s:PML,c:1622632123,t:LIVE*19\!AIVDM,1,1,,A,339K?fPP@ewe2jPLe7OP801F0000,0*4C
\s:PML,c:1622632123,t:LIVE*19\!AIVDM,1,1,,B,B3`p=Ph00GsAhuW<urAwWwn5CP06,0*1D
\s:PML,c:1622632123,t:LIVE*19\!AIVDM,1,1,,A,339K?fP00ewe2jPLe7OP801F0000,0*5C
\s:PML,c:1622632123,t:LIVE*19\!AIVDM,1,1,,A,339K?i001:wdoV0LjCf7:5cF20s0,0*4C
\s:PML,c:1622632124,t:LIVE*1E\!AIVDM,1,1,,B,B3MAat00?7s@LdW<pkQG7wn1h8JK,0*76
\s:PML,c:1622632124,t:LIVE*1E\!AIVDM,1,1,,A,13M@DFh000wdm9HLm5pTLEPn0@JO,0*0E
\s:PML,c:1622632124,t:LIVE*1E\!AIVDM,1,1,,A,B3P:skh00?sAK6W=6VgQ3wnQnDRb,0*60
\s:PML,c:1622632124,t:LIVE*1E\!AIVDM,1,1,,B,B3P=uwh00?sBPfW=43wQ3wnUoP06,0*66
\s:PML,c:1622632126,t:LIVE*1C\!AIVDM,1,1,,B,339K?fP00fwe2jVLe7a01h1L00w@,0*78
\s:PML,c:1622632126,t:LIVE*1C\!AIVDM,1,1,,B,339K?i001:wdoUlLjCIW8mcN20w1,0*08
\s:PML,c:1622632126,t:LIVE*1C\!AIVDM,1,1,,A,13M@BR001JOdmO>LlUlMm;7L00S6,0*69
\s:PML,c:1622632126,t:LIVE*1C\!AIVDM,1,1,,A,B3P7m>h0<Os02>7<cRR@gwoQl<0N,0*2F
\s:PML,c:1622632126,t:LIVE*1C\!AIVDM,1,1,,A,B3P9Nr00<wsDHh7;bovWcwoQl8Kv,0*3D
```

However if I try the following Python code:

```
#!/usr/bin/env python3
import socket
from pyais.stream import UDPStream

host = "192.171.162.68"
port = 12345

for msg in UDPStream(host, port):
    # msg.decode()
    print(msg)
    # do something with it
```

I get the error:

```
Traceback (most recent call last):
  File "/users/rsg/jpe/projects/pml-ais/log-roof-station.py", line 8, in <module>
    for msg in UDPStream(host, port):
  File "/users/rsg/jpe/.local/lib/python3.8/site-packages/pyais/stream.py", line 279, in __init__
    sock.bind((host, port))
OSError: [Errno 99] Cannot assign requested address
```

Also I tried the commandline utility and same issue:

```
jpe@pmpc1770:~>ais-decode socket -t udp 192.171.162.68 12345
Traceback (most recent call last):
  File "/usr/local/bin/ais-decode", line 8, in <module>
    sys.exit(main())
  File "/users/rsg/jpe/.local/lib/python3.8/site-packages/pyais/main.py", line 145, in main
    exit_code: int = namespace.func(namespace)
  File "/users/rsg/jpe/.local/lib/python3.8/site-packages/pyais/main.py", line 99, in decode_from_socket
    with stream_cls(args.destination, args.port) as s:
  File "/users/rsg/jpe/.local/lib/python3.8/site-packages/pyais/stream.py", line 279, in __init__
    sock.bind((host, port))
OSError: [Errno 99] Cannot assign requested address
```

My OS is Ubuntu 20.04 and Python 3.8.5. Does anyone know what the error is? Thank you

**Comment by pml-jpe** [comment]
OK I made some progress, if I run the AIS receiver in client mode and make my workstation act as server, and bind the socket to the `eno1` interface rather than localhost, it seems to work.

However due to the extra rubbish on each line that precedes the NMEA message, I get this error in Python:
```
ValueError: invalid literal for int() with base 10: b't:LIVE*13\\!AIVDM'
```

**Comment by M0r13n** [comment]
Hey there,

I guess that there is a bug:

```py
sock: socket = socket(AF_INET, SOCK_DGRAM)
sock.bind((host, port))
```

`socket.bind` only works for local interfaces. This is why, you got an `Errno 99` when trying to connect to an external IP.


I will look at this.

**Comment by M0r13n** [comment]
This issue will be resolved by upgrading to v2.0.0, which will be released in the next few weeks.
